Prepare for Renovate with reusable workflows

ghostwriter opened the pull request:
|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

Adding Renovate to Laminas, Mezzio repositories. https://gist.github.com/weierophinney/49ca8ae2b916d898dd3d4162b26ff41b

- Configure minimum supported PHP version
- Update composer dependencies
- Update to reusable `Continuous Integration` workflow
- Update to reusable `Automatic Releases` workflow

ghostwriter committed:
Configure minimum supported PHP version

Signed-off-by: Nathanael Esayeas <nathanael.esayeas@protonmail.com>

ghostwriter committed:
Update to reusable automatic-releases workflow

Signed-off-by: Nathanael Esayeas <nathanael.esayeas@protonmail.com>

ghostwriter committed:
Update to reusable continuous-integration workflow

Signed-off-by: Nathanael Esayeas <nathanael.esayeas@protonmail.com>

Ocramius reviewed (changes requested):
Requires rolling back to a much smaller `composer.lock` change (just hash change)

ghostwriter committed:
Fix tests

Status change of HTTP experiments to Historic: https://datatracker.ietf.org/doc/status-change-http-experiments-to-historic/

Signed-off-by: Nathanael Esayeas <nathanael.esayeas@protonmail.com>

ghostwriter committed:
Update composer.lock

Signed-off-by: Nathanael Esayeas <nathanael.esayeas@protonmail.com>

ghostwriter committed:
Update psalm.xml.dist

Signed-off-by: Nathanael Esayeas <nathanael.esayeas@protonmail.com>

ghostwriter committed:
Update psalm-baseline.xml

Signed-off-by: Nathanael Esayeas <nathanael.esayeas@protonmail.com>

weierophinney committed:
qa: resolve Psalm issues

Resolves the majority of issues flagged by Psalm.
The majority revolve around Psalm thinking a lowercase-string cannot be used for a non-empty-string, even when checks for non-empty strings are present.

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney committed:
qa: resolve CS issue

Long lines need to be ignored

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney commented:
The failing issue for Psalm... I cannot reproduce locally, so I cannot add it to the psalm baseline. Basically, it's raising an error because of this line in `Stream.php`:

```php
        if (PHP_VERSION_ID >= 80000 && $resource instanceof \GdImage) {
```

`GdImage` was one of the resource pseudo-classes introduced in PHP 8.0, and `instanceof` checks _do not need for the class to exist_ for the check to run. :angry:

weierophinney committed:
qa: suppress UndefinedClass Psalm error for GdImage

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>
